src/wp-includes/class-wp-theme.php

```diff
@@ -1506,6 +1525,36 @@ public function is_allowed( $check = 'both', $blog_id = null ) {
 		return false;
 	}
 
+
+	/**
+	 * Returns whether this theme has block template directories.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @return bool
+	 */
+	public function has_block_templates() {
+		if ( isset( $this->has_block_templates ) ) {
+			return $this->has_block_templates;
+		}
+
+		$paths_to_block_template_directories = array(
+			$this->theme_root . '/' . $this->template . '/block-template-parts',
+			$this->theme_root . '/' . $this->template . '/block-templates',
+		);
+
+		$this->has_block_templates = false;
+
+		foreach ( $paths_to_block_template_directories as $path_to_block_template_directory ) {
+			if ( is_dir( $path_to_block_template_directory ) ) {
+				$this->has_block_templates = true;
+				break;
+			}
+		}
+
+		return $this->has_block_templates;
+	}
```

Comment:
Based on my above suggestion, I'd suggest we change this to something like:
```suggestion
	/**
	 * Returns the folder names of the block template directories.
	 *
	 * @since 6.3.0
	 *
	 * @return string[] {
	 *     Folder names used by block themes.
	 *
	 *     @type string $wp_template      Theme-relative directory name for block templates.
	 *     @type string $wp_template_part Theme-relative directory name for block template parts.
	 * }
	 */
	public function get_block_template_folders() {
		if ( isset( $this->block_template_folders ) ) {
			return $this->block_template_folders;
		}

		$paths_to_block_template_directories = array(
			$this->theme_root . '/' . $this->template . '/block-template-parts',
			$this->theme_root . '/' . $this->template . '/block-templates',
		);

		$this->block_template_folders = array(
			'wp_template'      => 'templates',
			'wp_template_part' => 'parts',
		);

		foreach ( $paths_to_block_template_directories as $path_to_block_template_directory ) {
			if ( is_dir( $path_to_block_template_directory ) ) {
				$this->block_template_folders = array(
					'wp_template'      => 'block-templates',
					'wp_template_part' => 'block-template-parts',
				);
				break;
			}
		}

		return $this->block_template_folders;
	}
```

Reply:
Handled in https://github.com/WordPress/wordpress-develop/pull/5202